node/src/components/block_executor.rs

```diff
@@ -228,9 +232,7 @@ impl BlockExecutor {
                 deploys: result
                     .into_iter()
                     // Assumes all deploys are present
-                    .map(|maybe_deploy| {
-                        maybe_deploy.expect("deploy is expected to exist in the storage")
-                    })
+                    .map(|maybe_deploy| maybe_deploy.expect(&err_message))
```

Comment:
You could `unwrap_or_else(…)`, so the string only gets formatted and allocated if there's actually an error:
```rust
maybe_deploy.unwrap_or_else(|| panic!(
            "deploy for block in era={} and height={} is expected to exist in the storage",
            finalized_block.era_id(),
            finalized_block.height()
))
```

Reply:
This won't work b/c `finalized_block` is moved to `validate_block` method.

Reply:
Right; it would still be cheaper to just copy the era ID and height above and use it in this `panic!` instead of allocating the whole string above.

Reply:
ahh, Rust-afarians …

Reply:
Done.